nomos-services/blend/src/lib.rs

```diff
@@ -117,89 +126,306 @@ where
                 .relay::<MembershipService<EdgeService>>()
                 .await?,
             settings.crypto.signing_private_key.public_key(),
-        );
-        let mut _session_stream = SessionEventStream::new(
-            membership_adapter.subscribe().await?,
+        )
+        .subscribe()
+        .await?;
+        // TODO: Use membership_stream once the membership/SDP services are ready to provide the real membership: https://github.com/logos-co/nomos/issues/1532
+
+        let mut membership = settings.membership();
+        let mut session_stream = SessionEventStream::new(
+            Box::pin(constant_membership_stream(
+                membership.clone(),
+                settings.time.session_duration(),
+            )),
             settings.time.session_transition_period(),
         );
-        // TODO: Use session_stream: https://github.com/logos-co/nomos/issues/1532
-
-        // TODO: Add logic to start/stop the core or edge service based on the new
-        // membership info.
-
-        let minimal_network_size = settings.minimal_network_size;
-        let membership_size = settings.membership.len();
-
-        if membership_size >= minimal_network_size.get() as usize {
-            wait_until_services_are_ready!(
-                &overwatch_handle,
-                Some(Duration::from_secs(60)),
-                CoreService,
-                MembershipService<EdgeService>
-            )
-            .await?;
-            let core_relay =
-                overwatch_handle
-                .relay::<CoreService>()
-                .await?;
-            status_updater.notify_ready();
-            info!(
-                target: LOG_TARGET,
-                "Service '{}' is ready.",
-                <RuntimeServiceId as AsServiceId<Self>>::SERVICE_ID
-            );
-            while let Some(message) = inbound_relay.next().await {
-                debug!(target: LOG_TARGET, "Relaying a message to core service");
-                if let Err((e, _)) = core_relay.send(message).await {
-                    error!(target: LOG_TARGET, "Failed to relay message to core service: {e:?}");
-                }
-            }
-        } else {
-            wait_until_services_are_ready!(
-                &overwatch_handle,
-                Some(Duration::from_secs(60)),
-                NetworkService<NetworkBackendOfService<CoreService, RuntimeServiceId>, _>,
-                MembershipService<EdgeService>
-            )
-            .await?;
-            let core_network_relay =
-                overwatch_handle
+
+        // Relays to the core and edge services.
+        let core_relay = overwatch_handle.relay::<CoreService>().await?;
+        let edge_relay = overwatch_handle.relay::<EdgeService>().await?;
+
+        // A network adapter for broadcasting messages directly via gossipsub
+        // when the Blend network is too small.
+        let network_adapter = <CoreService::NetworkAdapter as NetworkAdapterTrait<
```

Comment:
This network adapter should also only ever exist if the network is too small. I think we can enforce these constraints at compile time so we don't misuse them. Or what was the reason to have them from the get-go instead? Anything I am not seeing?